lib/client.js

```diff
@@ -128,7 +152,7 @@ module.exports = function (dependencies) {
             resolve(this.write(notification, device, retryCount + 1));
             return;
           } else if (status === 500 && response.reason === "InternalServerError") {
-            this.session.destroy();
+            closeAndDestroy();
```

Comment:
The closeAndDestroy seems to be declared within an if statement. I'm not sure if it's callable - it may be referring to a global variable instead. You could use the tests in test/client.js to test if this works

Reply:
You're right. I will fix that.

Reply:
done